File: 2 тиждень/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
int keys(char a);

int main(int a, string b[])
{
    int zero = 0;
    if (a!=2)
    {
        printf("Usage: ./vigenere keyword\n");
        return 1;
    }
    for(int i=0, n=strlen(b[1]);i<n;i++)
    {
        if(b[1][i]<'A' || (b[1][i]>'Z' && b[1][i]<'a') || b[1][i]>'z')
        {
            printf("invalid keyword\n");
            return 1;
        }
    }
    int n=strlen(b[1]);
    int keyss[n];
    for(int i=0;i<n;i++)
    {
        keyss[i] = keys(b[1][i]);
    }
    string vvid = get_string("plaintext:  ");
    printf("ciphertext: ");
    int m = strlen(vvid);
    for(int i = 0;i<m;i++)
    {
        int chislo = vvid[i];
        if((chislo>=97&&chislo<=122)||(chislo>=65&&chislo<=90))
        {
            int ost1 = (i-zero)%n;
            if(chislo>=97&&chislo<=122)
            {
                chislo+=keyss[ost1];
                if(chislo>122)
                {
                    int ost = chislo-122;
                    chislo = 97+ost-1;
                    printf("%c",chislo);
                }
                else
                {
                    printf("%c",chislo);
                }
            }
            else 
            {
                chislo+=keyss[ost1];
                if(chislo>90)
                {
                    int ost = chislo-90;
                    chislo=65+ost-1;
                    printf("%c",chislo);
                }
                else
                {
                    printf("%c",chislo);

                }
            }
        }
        else
        {
            printf("%c",chislo);
            zero++;
        }
        
    }
    printf("\n");
}
int keys(char a)
{
    int b = a;
    if(b>=97&&b<=122)
    {     
        int c = b-97;
        return c;
    }
    else
    {
        int c = b-65;
        return c;
    }
}

File: 6 тиждень/vigenere/vigenere.py
import sys
from sys import argv
from cs50 import get_string


def main():
    if len(argv) != 2 or argv[1].isalpha() == False:
        print("Usage: python vigenere.py k ")
        sys.exit(1)
    vvid = get_string("plaintext:  ")
    i1 = 0
    print("ciphertext: ", end="")
    for i in range(len(vvid)):
        symbol1 = ord(vvid[i])
        k = (i-i1) % len(argv[1])
        if symbol1 >= 97 and symbol1 <= 122:
            symbol1 += hesh(argv[1][k])
            if symbol1 > 122:
                symbol1 -= 26
            print(chr(symbol1), end="")
        elif symbol1 >= 65 and symbol1 <= 90:
            symbol1 += hesh(argv[1][k])
            if symbol1 > 90:
                symbol1 -= 26
            print(chr(symbol1), end="")
        else:
            print(chr(symbol1), end="")
            i1 += 1
    print()
    return 0


def hesh(char):
    char1 = ord(char)
    if char1 >= 97 and char1 <= 122:
        return (char1-97)
    else:
        return (char1-65)


if __name__ == "__main__":
    main()